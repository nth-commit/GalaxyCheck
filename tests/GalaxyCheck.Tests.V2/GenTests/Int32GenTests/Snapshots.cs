using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
