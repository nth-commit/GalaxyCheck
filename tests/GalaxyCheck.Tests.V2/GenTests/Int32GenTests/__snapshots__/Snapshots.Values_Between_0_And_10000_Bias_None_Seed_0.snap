[
  0,
  1,
  1,
  0,
  1,
  1,
  0,
  1,
  1,
  0,
  1,
  1,
  0,
  1,
  0,
  0,
  1,
  1,
  1,
  1,
  4,
  0,
  4,
  4,
  1,
  3,
  1,
  4,
  1,
  4,
  3,
  1,
  4,
  0,
  0,
  4,
  2,
  1,
  0,
  2,
  3,
  9,
  8,
  7,
  5,
  0,
  4,
  3,
  1,
  8,
  10,
  10000,
  3,
  2,
  9,
  10,
  10,
  3,
  2,
  5,
  57,
  82,
  29,
  58,
  10,
  10000,
  64,
  3,
  36,
  75,
  56,
  96,
  93,
  50,
  10000,
  81,
  10000,
  94,
  10000,
  19,
  56,
  10000,
  1118,
  8228,
  6989,
  2793,
  8761,
  4128,
  10000,
  1296,
  276,
  1789,
  10000,
  10000,
  10000,
  10000,
  10000,
  5874,
  10000,
  10000
]
